### ISSUE TITLE
`aws-cdk-lib` package fails to process

### ISSUE THREAD

**Issue by addihorowitz** [cdklabs/construct-hub#626]
AWS CDK v2 fails to reprocess. 
Make sure it succeeds. 

It seems like it's one of the updates we've added to construct-hub 4? days ago. AWS CDK v2 version 31 was processed successfully but starting from 32 we're failing. 
I reviewed the code changes between 31 and 32 and it doesn't seem like dramatic changes were added. 

Perhaps we should consider reverting the latest code that was added?

**Comment by iliapolo** [comment]
Summarizing what we currently know about this issue. 

A while back, we started noticing that the ECS task for transliterating `aws-cdk-lib` was timing out. This is what sparked the following PR: [transliterator task times out on packages with many submodules](https://github.com/cdklabs/construct-hub/pull/617).

As mentioned in that PR, the reasons we suspected this was happening proved to be incorrect. Eventually, we landed on the assumption that because `aws-cdk-lib` recently added a magnitude of order more code samples to its inline documentation, it now takes a very long time for `jsii-rosetta` to [transliterate the assembly](https://github.com/cdklabs/jsii-docgen/blob/main/src/docgen/view/documentation.ts#L304). 

Since this transliteration is done on the main event loop, in a blocking way, it doesn't allow the heartbeat task to run and notify step functions that task is still alive, causing step functions to abandon it with a `States.Timeout` error. 

However, since then, `aws-cdk-lib` now includes rosetta tablet files in its distribution, dramatically reducing the work needed be done when transliterating the assembly. In addition, said transliteration underwent performance improvements and was moved to use a worker pool. 

All this work should have resolved the problem. It did not. `aws-cdk-lib` remains to fail, exhibiting the exact same symptoms.

And so, the investigation continues.
